x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/factories/wrap_hits_factory.ts

```diff
@@ -49,23 +53,35 @@ export const wrapHitsFactory =
         String(event._version),
         `${spaceId}:${completeRule.alertId}`
       );
+
+      const baseAlert = buildBulkBody(
+        spaceId,
+        completeRule,
+        event as SimpleHit,
+        mergeStrategy,
+        ignoreFields,
+        true,
+        buildReasonMessage,
+        indicesToQuery,
+        alertTimestampOverride,
+        ruleExecutionLogger
+      );
+
+      const alertUrl = getAlertDetailsUrl({
```

Comment:
It looks like we call `getAlertDetailsUrl` every time we call `buildBulkBody`, can we move the details URL logic inside `buildBulkBody`? That change will line up nicely with adding the `ALERT_URL` to `BaseFieldsLatest` since `buildBulkBody` returns `BaseFieldsLatest`.

Reply:
As we discussed on slack, I've updated the code so that the `build_alert.ts` has the logic for creating the url, and also sets the id. As suggested, I've updated all of the types and added a placeholder id for the eql alerts that's later overridden: https://github.com/elastic/kibana/pull/155069/commits/c916500cf6a7b8814de3467d6125210857e82ee8